src/representations/representation_manager_base.cc

```diff
@@ -30,9 +30,55 @@
 
 namespace rascal {
 
-  template<class StructureManager>
-  void RepresentationManagerBase::check_traits_compatibility(
-                                          StructureManager &) {
-    // Hello !
+  std::string RepresentationManagerBase::get_options_string() {
+    std::string out{};
+    for (const auto& item : this->options) {
+      out += item.second + std::string(" ");
+    }
+    if (not out.empty()) {
+      out.pop_back();
+    }
+    return out;
   }
-}
+  /* ---------------------------------------------------------------------- */
+  std::string RepresentationManagerBase::get_hypers_string() {
+    return this->hypers.dump(2);
+  }
+  /* ---------------------------------------------------------------------- */
+  void RepresentationManagerBase::check_hyperparameters(
```

Comment:
This function checks for presence of required options and for correct values of switches, but isn't this validation that we said we could do with json schemas? I can't remember our decision on using schemas, but they do have the additional benefit of type specification and validation of numerical values (that would be a pain to reinvent here).

Reply:
I don't remember either but I think that is would be good to keep such checks close to the implementation. Also having them too close to the json reader would mean a lot of work if we change/extend the reader capabilities.
In this function, I assume that the input to check has the same functionality as a stl map so it should allow for varying data representation.

Reply:
Fair enough -- in any case, it's good to keep it encapsulated in this check function so we can add or change functionality later as needed. It might still be a good idea for the future to align the semantics of the reference_hypers with that of the json schema, both so that it's familiar to schema users and so that we can switch later if needed. It's not a huge change; I think it's just instead of mapping a string to a list of strings, we map it to another map that contains fields such as `required` (boolean), `type` (str, int, enum, etc.), and `allowed_values` (or whatever it's called for a enum-like type).